### PR TITLE
Use zserio 2.5.0

### LINKED ISSUE #71: Transition to new zserio C++ reflection interface
As per zserio 2.5.0, there is a native zserio C++ reflection interface, so we can transition away from ZSR.

See https://github.com/ndsev/zserio/issues/103

### PR DESCRIPTION
**Pull request by josephbirkner**
Use cases of [zsr](https://github.com/kleber-engineering/zsr) have been replaced with official [zserio reflection](https://github.com/ndsev/zserio/issues/103) and a new [zserio-cmake-helper repo](https://github.com/kleber-engineering/zserio-cmake-helper). To compare what's changed from the user perspective, here are the old vs. new C++ integration examples from the README.

### New, using OAClient and native zserio

```cpp
#include "zswagcl/oaclient.hpp"
#include <iostream>
#include "myapp_services/services/MyService.h"

using namespace zswagcl;
using namespace httpcl;
using MyService = myapp_services::services::MyService;

int main (int argc, char* argv[])
{
    // Assume that the server provides its OpenAPI definition here
    auto openApiUrl = "http://localhost:5000/openapi.json";
    
    // Create an HTTP client to be used by our OpenAPI client
    auto httpClient = std::make_unique<HttpLibHttpClient>();
    
    // Fetch the OpenAPI configuration using the HTTP client
    auto openApiConfig = fetchOpenAPIConfig(specUrl, *httpClient);
    
    // Create a Zserio reflection-based OpenAPI client that
    // uses the OpenAPI configuration we just retrieved.
    auto openApiClient = OAClient(openApiConfig, std::move(httpClient));
        
    // Create a MyService client based on the OpenApi-Client
    // implementation of the zserio::IServiceClient interface.
    auto myServiceClient = MyService::Client(openApiClient);
    
    // Create the request object
    auto request = myapp_services::services::Request(2);

    // Invoke the REST endpoint
    auto response = myServiceClient.my_api(request);
    
    // Print the response
    std::cout << "Got " << response.getValue() << std::endl;
}
```

### Old, using ZSR

```cpp
#include "zswagcl/zsr-client.hpp"
#include <iostream>
#include <zsr/types.hpp>
#include <zsr/find.hpp>
#include <zsr/getset.hpp>

using namespace zswagcl;
using namespace httpcl;

int main (int argc, char* argv[])
{
    // Assume that the server provides its OpenAPI definition here
    auto openApiUrl = "http://localhost:5000/openapi.json";
    
    // Create an HTTP client to be used by our OpenAPI client
    auto httpClient = std::make_unique<HttpLibHttpClient>();
    
    // Fetch the OpenAPI configuration using the HTTP client
    auto openApiConfig = fetchOpenAPIConfig(specUrl, *httpClient);
    
    // Create a Zserio reflection-based OpenAPI client that
    // uses the OpenAPI configuration we just retrieved.
    auto zsrClient = ZsrClient(openApiConfig, std::move(httpClient));
        
    // Use reflection to find the service method that we want to call.
    auto serviceMethod = zsr::find<zsr::ServiceMethod>("services.MyService.my_api");
    
    // Use reflection to create the request object
    auto request = zsr::make(zsr::packages(), "services.Request", {{"value", 2}});

    // Invoke the REST endpoint
    auto response = serviceMethod->call(zsrClient, request);
    
    // Unpack the response variant as an introspectable struct 
    auto unpackedResponse = response.get<zsr::Introspectable>().value();
    
    // Use reflection to read the response's value member
    auto responseValue = zsr::get(unpackedResponse, "value").get<int>().value();
    
    // Print the response
    std::cout << "Got " << responseValue << std::endl;
}
```

Fixes #71

### CODE REVIEW

File: libs/zswag/test/client.cpp [hunk 1]
```diff
@@ -33,127 +31,96 @@ int main (int argc, char* argv[]) {
             auto openApiConfig = fetchOpenAPIConfig(specUrl, *httpClient);
             httpcl::Config authHttpConf;
             authFun(authHttpConf);
-            auto zsrClient = ZsrClient(openApiConfig, std::move(httpClient), authHttpConf);
+            auto zsrClient = OAClient(openApiConfig, std::move(httpClient), authHttpConf);
```
Comment (fklebert) [inline libs/zswag/test/client.cpp:34]:
```suggestion
            auto oaClient = OAClient(openApiConfig, std::move(httpClient), authHttpConf);
```

File: libs/zswag/test/client.cpp [hunk 2]
```diff
@@ -33,127 +31,96 @@ int main (int argc, char* argv[]) {
...
             std::cout << "[cpp-test-client]   → Running request." << std::endl;
-            auto responseObject = fn(zsrClient);
-            auto response = zsr::get(responseObject, "value").template get<decltype(expect)>().value();
-            if (response == expect)
+            calculator::Calculator::Client calcClient(zsrClient);
```
Comment (fklebert) [inline libs/zswag/test/client.cpp:36]:
```suggestion
            calculator::Calculator::Client calcClient(oaClient);
```

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 connexion
 requests
-zserio>=2.3.0
+zserio>=2.4.0rc2
```
Comment (fklebert) [inline requirements.txt:3]: Shouldn't we only use release versions?

Comment (josephbirkner) [reply requirements.txt:3]: Yes :)

**Review by fklebert — APPROVED** [review]

**Pull request merged**